cmd/vicadmin/vicadm.go

```diff
@@ -324,11 +326,25 @@ func listVMPaths(ctx context.Context, s *session.Session) ([]logfile, error) {
 			continue
 		}
 
-		logname, err := child.Name(ctx)
+		ec, err := child.FetchExtraConfig(ctx)
 		if err != nil {
-			log.Errorf("Unable to get the vm name for %s: %s", child.Reference(), err)
-			continue
+			log.Errorf("Unable to fetch the vm ExtraConfig for %s: %s", child.Reference(), err)
+
+			logname, err = child.Name(ctx)
+			if err != nil {
+				log.Errorf("Unable to get the vm name for %s: %s", child.Reference(), err)
+				continue
+			}
+		} else {
+			// for the appliance VM, id is in guestinfo.vice./init/common/id;
+			// however, this is not problematic since we don't collect these logs for the appliance VM
+			logname = ec[containerIDInExtraConfig]
```

Comment:
even the container id key is not supposed to be changed, using this way to get configuration is still not recommended.
vm.DSPath can return the vm datastore folder name, no need to query extraconfig here.